model_inference_triage_pipeline/run_model_inference.py

```diff
@@ -54,6 +55,10 @@ def main():
         s3_upload=S3_UPLOAD,
     )
 
+    # Save data to database using api server
+    if not cc.SKIP_INSERT_API_CALL:
+        save_data_to_db(start_time, end_time, CVE_MODEL_TYPE, S3_UPLOAD, ECOSYSTEM)
```

Comment:
`save_data_to_db` is violating the single-responsibility principle rule.

Instead of combining S3 upload on failure into single function, it would be better you do do that from here on failure of `save_data_to_db`.

Reply:
Done. Separated logic for saving failed records.